pkg/backend/ollama_backend.go

```diff
@@ -95,8 +116,9 @@ func (o *OllamaBackend) Generate(ctx context.Context, prompt string) (string, er
 	}
 	defer resp.Body.Close()
 
+	bodyBytes, _ := io.ReadAll(resp.Body)
+
 	if resp.StatusCode != http.StatusOK {
-		bodyBytes, _ := io.ReadAll(resp.Body)
```

Comment:
I think in another PR it would be nice to return structured errors so that the caller would do `errors.As` and would retrieve the body if needed (I agree having the body in-line as string is easier to debug in a tiny app)

Reply:
sounds good!